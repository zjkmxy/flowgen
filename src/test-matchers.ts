import { execFileSync } from "child_process";

import chalk from "chalk";
import flow from "flow-bin";

import { beautify } from ".";

// eslint-disable-next-line @typescript-eslint/no-unused-vars,no-unused-vars
declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace jest {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars,no-unused-vars
    interface Matchers<R> {
      toBeValidFlowTypeDeclarations(): R;
    }
  }
}

expect.extend({
  async toBeValidFlowTypeDeclarations(source) {
    const beautifiedSource = await beautify(source);
    try {
      execFileSync(
        flow,
        ["check-contents", "--color=always", "--timeout=30"],
        {
          input: beautifiedSource,
          stdio: ["pipe", "pipe", "pipe"],
        },
      );
    } catch (err) {
      return {
        message: () =>
          `expected ${chalk.bold(
            beautifiedSource.trimEnd(),
          )} to be valid flow:\n${chalk.red(err.stdout)}`,
        pass: false,
      };
    }

    return {
      message: () =>
        `expected ${chalk.bold(
          beautifiedSource.trimEnd(),
        )} not to be valid flow`,
      pass: true,
    };
  },
});
