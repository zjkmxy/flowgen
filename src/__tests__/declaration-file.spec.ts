import { compiler, beautify } from "..";
import "../test-matchers";

it("should handle basic keywords", async () => {
  const ts = `
declare interface A {
  bar: string
}

export declare const ok: number
  `;

  const result = compiler.compileDefinitionString(ts, {
    quiet: true,
    asModule: "test",
  });
  expect(await beautify(result)).toMatchSnapshot();
  expect(result).toBeValidFlowTypeDeclarations();
});

it("should handle basic keywords  cll", async () => {
  const ts = `
  declare module 'test' {
    interface A {
      bar: string
    }
    export const ok: number
  }`;

  const result = compiler.compileDefinitionString(ts, {
    quiet: true,
    asModule: "test",
  });
  expect(await beautify(result)).toMatchSnapshot();
  expect(result).toBeValidFlowTypeDeclarations();
});
