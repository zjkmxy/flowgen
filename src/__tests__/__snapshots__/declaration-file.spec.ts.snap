// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle basic keywords  cll 1`] = `
"declare module "test" {
  declare interface A {
    bar: string;
  }
  declare export var ok: number;
}
"
`;

exports[`should handle basic keywords 1`] = `
"declare module "test" {
  declare interface A {
    bar: string;
  }
  declare export var ok: number;
}
"
`;
